k8s/node.go

```diff
@@ -15,18 +15,19 @@ import (
 )
 
 const (
-	HostnameLabel             = "kubernetes.io/hostname"
-	InternalAddressAnnotation = "rke.cattle.io/internal-ip"
-	ExternalAddressAnnotation = "rke.cattle.io/external-ip"
-	AWSCloudProvider          = "aws"
-	MaxRetries                = 5
-	RetryInterval             = 5
+	HostnameLabel                = "kubernetes.io/hostname"
+	InternalAddressAnnotation    = "rke.cattle.io/internal-ip"
+	ExternalAddressAnnotation    = "rke.cattle.io/external-ip"
+	AWSCloudProvider             = "aws"
+	ExternalAWSCloudProviderName = "external-aws"
```

Comment:
Why can not we use the existing AWSCloudProvider? Can you explain the difference between those two?

Reply:
@jiaqiluo `external-aws` is same as `external` for Kubernetes. For RKE, when cloud provider is set to `external-aws` and `useInstanceMetadataHostname` is enabled, `rke-tools` will fetch the hostname by querying ec2 metadata service `http://169.254.169.254/latest/meta-data/hostname`. This name will be set as hostname-override to `kubelet` and `kube-proxy`.

We cannot make this amazon specific change under `external` as it's generic for all cloud providers. But it's required for node driver clusters to work (error in issue description), so `external-aws` is introduced.

Reply:
@kinarashah, what is the use case for the existing `aws`?  Why can we not use `aws` and `useInstanceMetadataHostname` to achieve the goal?

Reply:
@jiaqiluo That's a good idea and I think it'd have worked if we had kept it as `aws` and updated code to consider it as `external` for k8s>=1.27 + documented the behavior. Though from regression point of view and timing of the release, right now the change is all scoped under `external-aws` vs updating existing behavior for `aws` so it seems safer. 

Also this is not important but it would mean that existing k8s 1.26 clusters wouldn't be able to use this change because of k8s version checks with `aws` but they'd be able to migrate using `external-aws`.